### PR TITLE
[MM-40935] Add OnInstall() plugin hook

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR implements a `OnInstall()` hook that is triggered when the plugin is installed as part of the onboarding flow. A new type `model.OnInstallEvent` is used as an argument that allows future extension.

In the future, other install method like the Marketplace should also trigger the hook.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40935

#### Release Note

```release-note
Added OnInstall() plugin hook
```

### CODE REVIEW

**Comment by hanzei** [comment]
If the approach of this PR gets approval, I'm going to create tickets for the other install methods.

File: plugin/hooks.go [hunk 1]
```diff
@@ -249,4 +250,12 @@ type Hooks interface {
 	//
 	// Minimum server version: 6.4
 	RunDataRetention(nowTime, batchSize int64) (int64, error)
+
+	// OnInstall is invoked after the installation of a plugin as part of the unboarding.
```
Comment (catalintomai) [inline plugin/hooks.go:254]: unboarding - > onboarding

File: model/plugin_on_install_event.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+// OnInstallEvent is send to the plugin when it gets installed.
+type OnInstallEvent struct {
```
Comment (catalintomai) [inline model/plugin_on_install_event.go:7]: do we expect to have more events similar to this in the future - if yes, would it make sense to have an `events` file instead?

**Review by catalintomai — APPROVED** [review]

File: model/plugin_on_install_event.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+// OnInstallEvent is send to the plugin when it gets installed.
```
Comment (catalintomai) [inline model/plugin_on_install_event.go:6]: send -> sent

File: plugin/hooks.go [hunk 4]
```diff
@@ -249,4 +250,12 @@ type Hooks interface {
...
+	// It's called on every installation, not only once.
+	//
+	// In the future, other plugin installation method will trigger this hook, e.g. an installation via the Marketplace.
```
Comment (catalintomai) [inline plugin/hooks.go:257]: method -> method(s)

File: model/plugin_on_install_event.go (hunk 2 above)
Comment (hanzei) [reply model/plugin_on_install_event.go:7]: Yes, I would expect more plugin events to be added. https://github.com/mattermost/mattermost-server/blob/97ccf0bdf6f407d7308d58763f7687d3f079db67/model/plugin_cluster_event.go is the other existing one.

0/5 if it makes sense to consolidate into a single file. They are "stand alone" events and have nothing in common with each other besides that they are events.

**Review by levb — APPROVED** [review]
👍

**Review by DHaussermann — APPROVED** [review]
@hanzei This would be far more efficient to test post merge. 
For now there is not much we can do for functional testing on this.

We should test this post-merge.

File: model/plugin_on_install_event.go (hunk 2 above)
Comment (catalintomai) [reply model/plugin_on_install_event.go:7]: the idea was to avoid creating one (few lines) file per event if we expect (many) more events in the future.

**Pull request merged**